File: Tasks/factory/TaskContainerFactory.cs
using Lab10.containers;

namespace Lab10.factory;

public class TaskContainerFactory : Factory
{
    //Implementeaza interfata Factory folosind principiul Singleton
    
    private static TaskContainerFactory instance;
    
    //Constructor
    private TaskContainerFactory() { }
    
    //Metoda pentru a obtine instanta clasei
    public static TaskContainerFactory getInstance()
    {
        if (instance == null)
        {
            instance = new TaskContainerFactory();
        }
        return instance;
    }

    //Implementeaza metoda createContainer din interfata Factory
    public Container createContainer(ContainerStrategy strategy)
    {
        if (strategy.Equals(ContainerStrategy.FIFO))
        {
            return new QueueContainer();
        }
        else
        {
            return new StackContainer();
        }
    }
}

File: Tasks/src/factory/TaskContainerFactory.java
package factory;

import model.Container;
import model.QueueContainer;
import model.StackContainer;

public class TaskContainerFactory implements Factory {
    //Implementeaza interfata Factory folosind principiul Singleton

    private static TaskContainerFactory instance;

    //Constructor
    private TaskContainerFactory() {}

    //Metoda pentru a obtine instanta unei clase
    public static TaskContainerFactory getInstance(){
        if(instance == null){
            instance = new TaskContainerFactory();
        }
        return instance;
    }

    //Suprascrie metoda createContainer() din interfata Factory
    @Override
    public Container createContainer(Strategy strategy) {
        if(strategy == Strategy.FIFO){ //Creeaza si returneaza un container de tip FIFO
            return new QueueContainer();
        } else { //Creeaza si returneaza un container de tip LIFO
            return new StackContainer();
        }
    }
}
